builtin/providers/google/resource_sql_database_instance.go

```diff
@@ -0,0 +1,948 @@
+						}
+					}
+				}
+
+				if vp, okp := _ipConfiguration["authorized_networks"]; okp || len(_oldAuthorizedNetworkList) > 0 {
+					oldAuthorizedNetworks := settings.IpConfiguration.AuthorizedNetworks
+					settings.IpConfiguration.AuthorizedNetworks = make([]*sqladmin.AclEntry, 0)
+
+					_authorizedNetworksList := make([]interface{}, 0)
+					if vp != nil {
+						_authorizedNetworksList = vp.([]interface{})
+					}
+					_oipc_map := make(map[string]interface{})
+					for _, _ipc := range _oldAuthorizedNetworkList {
+						_entry := _ipc.(map[string]interface{})
+						_oipc_map[_entry["value"].(string)] = true
+					}
+					// Next read the network tuples from the server, and reinsert those that
+					// were not previously defined
+					for _, entry := range oldAuthorizedNetworks {
+						_, ok_old := _oipc_map[entry.Value]
+						if !ok_old {
+							settings.IpConfiguration.AuthorizedNetworks = append(
+								settings.IpConfiguration.AuthorizedNetworks, entry)
+						}
+					}
+					// finally, insert only those that were previously defined
+					// and are still defined.
+					for _, _ipc := range _authorizedNetworksList {
+						_entry := _ipc.(map[string]interface{})
+						if _, ok_old := _oipc_map[_entry["value"].(string)]; ok_old {
+							entry := &sqladmin.AclEntry{}
+
+							if vpp, okpp := _entry["expiration_time"]; okpp {
+								entry.ExpirationTime = vpp.(string)
+							}
+
+							if vpp, okpp := _entry["name"]; okpp {
+								entry.Name = vpp.(string)
+							}
+
+							if vpp, okpp := _entry["value"]; okpp {
+								entry.Value = vpp.(string)
+							}
+
+							settings.IpConfiguration.AuthorizedNetworks = append(
+								settings.IpConfiguration.AuthorizedNetworks, entry)
+						}
+					}
+				}
+			}
+		}
+
+		if v, ok := _settings["location_preference"]; ok {
+			_locationPreferenceList := v.([]interface{})
+			if len(_locationPreferenceList) > 1 {
+				return fmt.Errorf("At most one location_preference block is allowed")
+			}
+
+			if len(_locationPreferenceList) == 1 && _locationPreferenceList[0] != nil {
+				settings.LocationPreference = &sqladmin.LocationPreference{}
+				_locationPreference := _locationPreferenceList[0].(map[string]interface{})
+
+				if vp, okp := _locationPreference["follow_gae_application"]; okp {
+					settings.LocationPreference.FollowGaeApplication = vp.(string)
+				}
+
+				if vp, okp := _locationPreference["zone"]; okp {
+					settings.LocationPreference.Zone = vp.(string)
+				}
+			}
+		}
+
+		if v, ok := _settings["pricing_plan"]; ok {
+			settings.PricingPlan = v.(string)
+		}
+
+		if v, ok := _settings["replication_type"]; ok {
+			settings.ReplicationType = v.(string)
+		}
+
+		instance.Settings = settings
+	}
+
+	d.Partial(false)
+
+	op, err := config.clientSqlAdmin.Instances.Update(config.Project, instance.Name, instance).Do()
+	if err != nil {
+		return fmt.Errorf("Error, failed to update instance %s: %s", instance.Name, err)
```

Comment:
better to say "google_sql_database_instance" because otherwise it looks like a compute engine instance